src/tile.js

```diff
@@ -282,14 +287,29 @@ export default class Tile {
     // TODO: pass bounds only, rest of scene isn't needed
     updateVisibility(scene) {
         var visible = this.visible;
-        this.visible = this.isInZoom(scene) && Geo.boxIntersect(this.bounds, scene.bounds_meters_buffered);
+        this.visible = this.isInZoom(scene.capped_zoom) && Geo.boxIntersect(this.bounds, scene.bounds_meters_buffered);
         this.center_dist = Math.abs(scene.center_meters.x - this.min.x) + Math.abs(scene.center_meters.y - this.min.y);
         return (visible !== this.visible);
     }
 
     // TODO: pass zoom only?
-    isInZoom(scene) {
-        return (Math.min(this.coords.z, this.tile_source.max_zoom || this.coords.z)) === scene.capped_zoom;
+    isInZoom(zoom) {
+        return (Math.min(this.coords.z, this.max_zoom || this.coords.z)) === zoom;
+    }
+
+    // TODO fix the z adjustment for continuous zoom
+    calculateOverZoom() {
```

Comment:
Hmm... overzoom really is a source-specific thing. Let's discuss how to keep it there.